### DISCUSSION TITLE
Save data from Pi to NAS

### DISCUSSION THREAD

**Discussion by abd7786** [jasonacox/Powerwall-Dashboard#351] [General]
Hi
Planning on using a Raspberry Pi 4gb with 64gb sd card to run the powewall dashboard.

With the continuous writing of logged data i fear that the micro SD card may give up.
Instead of purchasing a ssd, i already have a NAS on the network with no password for a certain folder.

If I mount the NAS storage on the PI using [this](http://cagewebdev.com/raspberry-pi-connecting-to-a-network-drive/) , would the powerwall-dashboard work fine to write data to the NAS rather than the sd?

As a newbie to coding on the Pi, how could this  easily be achieveable?

Thanks!

**Comment by jasonacox** [comment]
Good question.  I think there are a few things here:

## Database

The mounting of a NAS device is a great idea, espeically if the NAS is set up with data guarding (RAID level etc) - If the RPi or the SD card fails, you can easily rebuild one and mount the drive. I assume your idea would be to mount a NAS share to `/home/pi` or some mount point where you would install the Powerwall-Dashboard

## Logs

The docker logs are actually recorded in `/var/lib/docker/containers/*/*-json.log` - Which means you could use `/var/lib/docker/containers` as the mount point for your NAS, you could update docker on your system to store those logs in a different location or turn them off all together.

I asked ChatGPT and it gave some help on how to adjust docker, either to disable logs or store them in a different location (e.g. a NAS): https://chat.openai.com/share/51319da2-989a-46e8-802b-6f6b06bf7e89 

* Turn them off - Edit `/lib/systemd/system/docker.service` and `--log-driver=none` to the ExecStart line.

* Restart: 

   `sudo systemctl daemon-reload`
   `sudo systemctl restart docker`
